library/data/component/src/main/java/org/quiltmc/qsl/component/api/components/NbtComponent.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.component.api.components;
+
+import net.minecraft.nbt.*;
+import net.minecraft.util.Identifier;
+import org.jetbrains.annotations.Nullable;
+import org.quiltmc.qsl.component.api.Component;
+import org.quiltmc.qsl.component.impl.util.ErrorUtil;
+
+public interface NbtComponent<T extends NbtElement> extends Component {
+	@SuppressWarnings("unchecked")
+	static void readFrom(NbtComponent<?> nbtComponent, Identifier id, NbtCompound root) {
+		NbtElement nbtTarget = root.get(id.toString());
+		switch (nbtComponent.nbtType()) {
+			case NbtElement.BYTE_TYPE -> ((NbtComponent<NbtByte>) nbtComponent).read((NbtByte) nbtTarget);
+			case NbtElement.SHORT_TYPE -> ((NbtComponent<NbtShort>) nbtComponent).read((NbtShort) nbtTarget);
+			case NbtElement.INT_TYPE -> ((NbtComponent<NbtInt>) nbtComponent).read((NbtInt) nbtTarget);
+			case NbtElement.LONG_TYPE -> ((NbtComponent<NbtLong>) nbtComponent).read((NbtLong) nbtTarget);
+			case NbtElement.FLOAT_TYPE -> ((NbtComponent<NbtFloat>) nbtComponent).read((NbtFloat) nbtTarget);
+			case NbtElement.DOUBLE_TYPE -> ((NbtComponent<NbtDouble>) nbtComponent).read((NbtDouble) nbtTarget);
+			case NbtElement.BYTE_ARRAY_TYPE ->
```

Comment:
The line breaks here seems way too arbitrary on this wall of text (oh, and on a kinda unrelated note: I love switch expressions);
Yes yes, there's a char limit per line and such; Ignore 'em here, remove the line breaks on BYTE_ARRAY_TYPE and LONG_ARRAY_TYPE; These limits do make sense as a style guide and you should follow them when it makes sense, but strictly enforcing them just because it passed few characters from it doesn't make sense (heck, iiirc, Linus Torvalds complained about the strictness of that)

Reply:
IntelliJ autoformatting got me on those. I actually just had them all be one liners.